draft-ietf-mls-architecture.md

```diff
@@ -696,6 +696,109 @@ software is updated while the client is a member of a group. Thus, in addition
+    sync with the state of the group.
+  - If an application chooses not to allow assisted or external joining, it may
+    instead provide a method for external users to solicit group members (or a
+    designated service) to add them to a group.
+  - If the application uses external PSKs, or uses resumption PSKs that all
+    members of a group may not have access to, there must be a method for
+    distributing these PSKs to group members.
+  - If an application wishes to detect and possibly discipline members that send
+    malformed commits with the intention of corrupting a group's state, there
+    must be a method for reporting and validating malformed commits.
+
+MLS requires the following parameters to be defined, which must be the same for
+two implementations to interoperate:
+
+- The maximum total lifetime that is acceptable for a KeyPackage.
+- How long to store the resumption secret for past epochs of a group.
+- The degree of tolerance that's allowed for out-of-order message delivery:
+  - How long to keep unused nonce and key pairs for a sender
+  - A maximum number of unused key pairs to keep.
+  - A maximum number of steps that clients will move a secret tree ratchet
+    forward in response to a single message before rejecting it.
+
+MLS provides the following locations where an application may store arbitrary
+data. The format and intention of any data in these locations must align for two
+deployments to interoperate:
+
+- Application data, sent as the payload of an encrypted message.
+- Additional authenticated data, sent unencrypted in an otherwise encrypted
+  message.
+- Group IDs, as decided by group creators and used to uniquely identify a group.
+- The `application_id` extension of a LeafNode.
+
+MLS requires the following policies to be defined, which restrict the set of
+acceptable behavior in a group. These policies must be consistent between
+deployments for them to interoperate:
+
+- A policy on when to send proposals and commits in plaintext instead of
+  encrypted.
+- A policy for which proposals are valid to have in a commit, including but not
+  limited to:
+  - When a member is allowed to add or remove other members of the group.
+  - When, and under what circumstances, a reinitialization proposal is allowed.
+  - When proposals from external senders are allowed.
+  - When external joiners are allowed.
+- A policy for when two credentials represent the same client. Note that many
+  credentials may be issued authenticating the same identity but for different
+  signature keys, because each credential corresponds to a different device
+  (client) owned by the same application user. However, one device may control
+  many signature keys but should still only be considered a single client.
+- A policy on how long to allow a member to stay in a group without updating its
+  leaf keys, and whether there should be any attempt to resolve a stale leaf key
+  before removing the member.
```

Comment:
What do you mean "resolve a stale leaf key"?  Like prompt the client to update?  I would just remove this last clause, or amend to something like "... and what actions should be taken with regard to stale leaf keys".